### PR TITLE
Feat/random greeting

### PR DESCRIPTION
**Pull request by cancelei**
Hello reviewer!

- Implement a connection between a Ruby on Rails back-end and React front-end.
- Post gress database (Using .env file, that one needs to fill in to use.)
- API endpoint that generates greetings.
- Static front end that is pointed to the root.

Happy review!

### CODE REVIEW

File: app/javascript/components/GreetingContext.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  useEffect(() => {
+    async function fetchGreeting() {
+      const response = await fetch('/greetings/random');
+      const data = await response.json();
+      setGreeting(data.message);
+    }
+
+    fetchGreeting();
+  }, []);
+
```
Comment (OLIPLICHE) [inline app/javascript/components/GreetingContext.js]: - [ ] Hi, You have done a great job in displaying random Greating, But I believe that you did not follow well the requirements. Learning objectives here is to **implement a connection between a Ruby on Rails back-end and a React front-end.** 
- The requirement says: **Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.**  I would advise you to follow this requirement.
- The picture below is the feedback I got when I changed the API in your react app to your localhost (a connection between a Ruby on Rails back-end and a React front-end)
Capture

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @cancelei, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Used Webpack with jsbundling-rails 👍 
- Created Rails app 👍 
- The project has a Postgres database 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @cancelei, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well adding a reducer, store and action creators as requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — COMMENTED** [review]
Hi @cancelei, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**